src/sentry/models/event.py

```diff
@@ -320,75 +310,310 @@ def site(self):
+        return self.__dict__.get('type', 'default')
+
+    # These should all have been normalized to the correct values on
+    # the way in to snuba, so we should be able to just use them as is.
+    @property
+    def ip_address(self):
+        return self.__dict__['ip_address']
+
+    @property
+    def title(self):
+        return self.__dict__['title']
+
+    @property
+    def culprit(self):
+        return self.__dict__['culprit']
+
+    @property
+    def location(self):
+        return self.__dict__['location']
+
+    # ============================================
+    # Snuba implementations of django Fields
+    # ============================================
+    @property
+    def datetime(self):
+        """
+        Reconstruct the datetime of this event from the snuba timestamp
+        """
+        # dateutil seems to use tzlocal() instead of UTC even though the string
+        # ends with '+00:00', so just replace the TZ with UTC because we know
+        # all timestamps from snuba are UTC.
+        return parse_date(self.timestamp).replace(tzinfo=pytz.utc)
+
+    @property
+    def time_spent(self):
+        return None
+
+    @property
+    def id(self):
+        # Because a snuba event will never have a django row id, just return
+        # the hex event_id here. We should be moving to a world where we never
+        # have to reference the row id anyway.
+        return self.event_id
+
+    def next_event(self, environments=[]):
+        from sentry.utils import snuba
+
+        conditions = [['event_id', '!=', self.id]]
+
+        if len(environments) > 0:
+            conditions.append(['environment', 'IN', environments])
+
+        result = snuba.raw_query(
+            start=self.datetime,  # gte current event
+            end=datetime.utcnow(),  # will be clamped to project retention
+            selected_columns=self.selected_columns,
+            conditions=conditions,
+            filter_keys={
+                'project_id': [self.project_id],
+                'issue': [self.group_id],
+            },
+            orderby=['timestamp', 'event_id'],
+            limit=5
+        )
+
+        if 'error' in result or len(result['data']) == 0:
+            return None
+
+        events = [e for e in result['data'] if
+                  e['timestamp'] == self.timestamp and
+                  e['event_id'] > self.event_id or
+                  e['timestamp'] > self.timestamp]
+
+        return SnubaEvent(events[0]) if events else None
+
+    def prev_event(self, environments=None):
+        from sentry.utils import snuba
+
+        conditions = [['event_id', '!=', self.id]]
+
+        if len(environments) > 0:
+            conditions.append(['environment', 'IN', environments])
+
+        result = snuba.raw_query(
+            start=datetime.utcfromtimestamp(0),  # will be clamped to project retention
+            end=self.datetime,  # lte current event
+            selected_columns=self.selected_columns,
```

Comment:
If I understand correctly, we only need the event_id here to put in the 'next' field. We can return the full set of columns later when the user actually clicks on the next link.  Adding all the `selected columns` here is going to make clickhouse look at a lot more columns than it needs to, and will make this query a lot slower.

Reply:
Good point, updated these functions to be `next_event_id` and `prev_event_id`

Reply:
It seems the event details endpoint is currently returning the database ids instead of `event_ids` for next/prev id values.
https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/event_details.py#L38-L39

As a result of this we'll also be swapping these out for event_ids there then

Reply:
yeah I think we should move to have the event details endpoint just return `event_id`s for next and prev events, since the details endpoint supports looking up events both ways, it should just work 🤞